src/libexpr/eval-error.cc

```diff
@@ -101,5 +102,12 @@ template class EvalErrorBuilder<MissingArgumentError>;
 template class EvalErrorBuilder<InfiniteRecursionError>;
 template class EvalErrorBuilder<CachedEvalError>;
 template class EvalErrorBuilder<InvalidPathError>;
+template class EvalErrorBuilder<PoisonContextError>;
+
+PoisonContextError::PoisonContextError(EvalState & state)
+    : EvalError(state, "Found 'poison' context that may not be built or included in derivations")
```

Comment:
Are more descriptive message would be helpful; perhaps one stored in the context.